cpp/src/gandiva/precompiled/extended_math_ops_test.cc

```diff
@@ -87,4 +87,13 @@ TEST(TestExtendedMathOps, TestLogWithBase) {
   EXPECT_EQ(context1.has_error(), false);
 }
 
+TEST(TestExtendedMathOps, TestTruncate) {
+  EXPECT_EQ(truncate_int64_int32(1234, 4), 1234);
+  EXPECT_EQ(truncate_int64_int32(-1234, 4), -1234);
+  EXPECT_EQ(truncate_int64_int32(1234, -4), 0);
+  EXPECT_EQ(truncate_int64_int32(-1234, -2), -1200);
+  EXPECT_EQ(truncate_int64_int32(8124674407369523212, 0), 8124674407369523212);
+  EXPECT_EQ(truncate_int64_int32(8124674407369523212, -2), 8124674407369523200);
```

Comment:
```suggestion
  EXPECT_EQ(truncate_int64_int32(8124674407369523212, -2), 8124674407369523200);
  EXPECT_EQ(truncate_int64_int32(8124674407369523212, 2), 8100000000000000000);
```

Reply:
for positive second argument, truncation on integers will have no effect
(see https://en.wikipedia.org/wiki/Truncation)